tests/blackbox.py

```diff
@@ -282,31 +285,36 @@ def program(self) -> Expr:
 
         return self._pyteal_lambda()
 
-    def _handle_SubroutineFnWrapper(self):
+    def _arg_prep_n_call(self, i, p):
         subdef = self.subr.subroutine.subroutine
         arg_names = subdef.arguments()
+        name = arg_names[i]
+        arg_expr = Txn.application_args[i] if self.mode == Mode.Application else Arg(i)
+        if p == TealType.uint64:
+            arg_expr = Btoi(arg_expr)
+        prep = None
+        arg_var = arg_expr
+        if name in subdef.by_ref_args:
+            arg_var = ScratchVar(p)
+            prep = arg_var.store(arg_expr)
+        elif name in subdef.abi_args:
+            arg_var = p.new_instance()
+            prep = arg_var.decode(arg_expr)
```

Comment:
```suggestion
        if name in subdef.by_ref_args:
            arg_var = ScratchVar(p)
            prep = arg_var.store(arg_expr)
        elif name in subdef.abi_args:
            arg_var = p.new_instance()
            prep = arg_var.decode(arg_expr)
        else:
            arg_var = arg_expr
            prep = None
```

slightly more readable (should have probably done this in my previous version)

Reply:
good point